include/deal.II/lac/linear_operator.h

```diff
@@ -1111,6 +1111,130 @@ linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
   return return_op;
 }
 
+template <typename Range = Vector<double>,
+          typename Domain = Range,
+          typename Matrix>
+LinearOperator<Range, Domain>
+constraints_linear_operator(const ConstraintMatrix &, const Matrix &);
+
+/**
+ * @relates LinearOperator
+ *
+ * constraints_linear_operator takes a constraint matrix @p cm and an exemplar
+ * matrix @p m (this matrix could be the system matrix).
+ * The result of this function is the LinearOperator associated to the constraint
+ * matrix.
+ * In detail, a constrained problem can be expressed in this form:
+ * Ct * S * C * x = Ct * ( rhs + S * k)
+ * Where:
+ *  - C has the role of constraint matrix
+ *  - k is the vector representing constraints inhomogeneities.
+ *  [ a generic contraint can be expressed as x = Cx + k ]
+ *  - S is the system matrix
+ *  - rhs is the original right-hand-side
+ * This function returns a LinearOperator representing the matrix C.
+ *
+ * [ see M. S. Shephard:. Linear multipoint constraints applied via
+ * transformation as part of a direct stiffness assembly process., 1985.
+ * for more details ]
```

Comment:
Your periods '.' are in the wrong place. There shouldn't be one after the author + colon. There shouldn't be one after the title. There should be a comma after 1985. And there should be one after "details".

Instead of putting this into parentheses, just use `@see` to mark it up for doxygen.